File: lab3.py
import requests
import re
import time
file=open('result.html', 'w')
file.write('<!DOCTYPE html>\n')
file.write('<html><head><meta http-equiv="Content-Type" content="text/html" charset=UTF-8"></head><body>')
now=int(time.time())*1000

country = "us"
category_id="34"
city="New York"
state="NY"




r=requests.get('https://api.meetup.com/2/concierge?key=687717166a6b43701f1c6b103822620&sign=true&photo-host=public&country='+country+'&city='+city+'&category_id='+category_id+'&state='+state)
name=re.findall(r'"name":"([\w\d\ \(\)\:\&\,\.\@\!\#\'\+\/\-]+)","id":"[\w ]+","time"', r.text)
adress_l=re.findall(r'"address_1":"([\w\ \.\&]+)"', r.text)
t=re.findall(r'"name":"[\w\d\ \(\)\:\&\,\.\@\!\#\'\+\/\-]+","id":"[\w ]+","time":([\d]+)', r.text)
print(name)
print(adress_l)
print(t)
for i in range(7):
    file.write( str(i+1) +' day: \n<ul>\n') #1 день с запуска программы +24 часа. 2ой день с +24 до +48
    for j in range(len(t)):
        if (int(t[j])<((i+1)*86400000+now)) and (int(t[j])>(i*86400000+now)): #86400000 это 24 часа ибо сутки надо выделять.
            t2=time.strftime("%a, %d %b %Y %H:%M:%S", time.localtime(float(t[j])/1000-43200))   #43200 это 12 часов ибо разница с городом в 14 часов
            file.write('<li>' + str(t2) + ',<br> ' + str(name[j]) + ',<br> ' + str(adress_l[j]) + '</li>\n')
    file.write('</ul>\n')
file.write('</body></html>')

File: lab3.js
var request = require('request');
var fs = require('fs');

var country = "us"
var category_id="34"
var city="New York"
var state="NY"
var url = "https://api.meetup.com/2/concierge?key=687717166a6b43701f1c6b103822620&sign=true&photo-host=public&country=" + country + '&city=' + city + '&category_id=' + category_id + '&state=' + state;

var adressRegex = /^$|"address_1":"([\w\ \.\&]+)"/igm;
var eventsGerex = /"name":"[\w\d\ \(\)\:\&\,\.\@\!\#\'\+\/\-]+","id":"[\w ]+","time":([\d]+)/ig;

request(url, function (err,res,body) 
{
    if (err) {
        //throw err;
    }
    else {
        var events = JSON.parse(body).results;
        var now = new Date().getTime();
        var eventsMarkup = "<body><html>";

        for (j = 0; j < 7; j++){
            for (i = 0; i < events.length; i++) {
                if (events[i].time<((j+1)*86400000+now) && events[i].time>(j*86400000+now)) {
                    var date = new Date(events[i].time);
                    var year = date.getFullYear();
                    var months = ['Янв', 'Фев', 'Мар', 'Апр', 'Май', 'Июн', 'Июл', 'Авг', 'Сен', 'Окт', 'Ноя', 'Дек'];
                    var month = months[date.getMonth()];
                    var date = date.getDate();
                    var formattedTime = year + '-' + month + '-' + date;
                    
                    var eventHtml = "not";
                    if (typeof events[i].venue === 'undefined') {
                        var eventHtml = '<li>' + formattedTime + ',<br> ' + events[i].name + ',<br> ' + "No address" + '</li>\n';
                    }
                    else {
                        var eventHtml = '<li>' + formattedTime + ',<br> ' + events[i].name + ',<br> ' + events[i].venue.address_1 + '</li>\n';
                    }
                    eventsMarkup += eventHtml;
                }
            }
        }
    }

    eventsMarkup += '</body></html>';
    fs.writeFile("events.html", eventsMarkup, function(err) {
        if(err) {
            return console.log(err);
        }
        console.log("The file was saved!");
    }); 
});